bot.py

```diff
@@ -96,44 +94,50 @@ async def get_rank(ctx, name: str):
-        summoner_name_opgg = summoner_name.replace(" ", "")
-
-        # Add author, thumbnail, fields, and footer to the embed
-        embed.set_author(
-            name=summoner_name,
-            url=f"https://na.op.gg/summoner/userName={summoner_name_opgg}",
-            icon_url=summoner_info["summoner_icon_image_url"],
+        embed_data = EmbedData()
+        embed_data.title = "Solo/Duo Rank"
+        embed_data.description = (
+            f"`All Data from NA server`\n\n <@!{bot.user.id}> <command>"
         )
+        embed_data.color = discord.Color.dark_gray()
 
-        # Get image of tier by path
-        tier_image = summoner_info["tier_image"]
-        file = discord.File(tier_image)
-        # Need to get the filename in order to attach to the thumbnail
+        # Add author, thumbnail, fields, and footer to the embed
+        embed_data.author = {}
+        embed_data.author = {
+            "name": summoner_info["user_name"],
+            # For op.gg link, we have to remove all whitespace.
+            "url": "https://na.op.gg/summoner/userName={0}".format(
+                summoner_info["user_name"].replace(" ", "")
+            ),
+            "icon_url": summoner_info["summoner_icon_image_url"],
+        }
+
+        # Upload tier image to discord to use it as thumbnail of embed using full path of image.
+        file = discord.File(summoner_info["tier_image_path"])
 
-        tier_image_filename = tier_image.replace("ranked-emblems/", "")
         # Embed thumbnail image of tier at the side of the embed
-        embed.set_thumbnail(url=f"attachment://{tier_image_filename}")
+        # Note: This takes the 'file name', not a full path.
+        embed_data.thumbnail = "attachment://{0[tier_image_name]}".format(summoner_info)
 
         # Setting variables for summoner information to display as field
-        summoner_rank = summoner_info["tier"]
-        solo_win = summoner_info["solo_win"]
-        solo_loss = summoner_info["solo_loss"]
-        summoner_total_game = solo_win + solo_loss
-        solo_rank_percentage = int(solo_win / summoner_total_game * 100)
-
-        embed.add_field(
-            name=summoner_rank,
-            value="Total Games Played:"
-            + f"{summoner_total_game}\n{solo_win}W \
-            {solo_loss}L {solo_rank_percentage}%",
-            inline=False,
+        summoner_total_game = summoner_info["solo_win"] + summoner_info["solo_loss"]
+        solo_rank_win_percentage = int(
+            summoner_info["solo_win"] / summoner_total_game * 100
+        )
+
+        embed_data.fields = []
+        embed_data.fields.append(
+            {
+                "name": "{0[tier]}".format(summoner_info),
+                "value": "Total Games Played: {1}\n{0[solo_win]}W {0[solo_loss]}L {2}%".format(
+                    summoner_info,
+                    summoner_total_game,
+                    solo_rank_win_percentage,
+                ),
```

Comment:
same here. 
With this formatting, no need to declare 
```
summoner_rank = summoner_info["tier"]
solo_win = summoner_info["solo_win"]
solo_loss = summoner_info["solo_loss"]
```
like the old way

Reply:
This is good learning for me too! Thanks!